libbeat/outputs/elasticsearch/client.go

```diff
@@ -154,20 +158,44 @@ func NewClient(
 		return nil
 	}
 
-	// Make sure there's a non-nil obser
+	// Make sure there's a non-nil observer
 	observer := s.observer
 	if observer == nil {
 		observer = outputs.NewNilObserver()
 	}
 
+	log := logp.NewLogger("elasticsearch")
+
+	pLogDeadLetter := periodic.NewDoer(10*time.Second,
```

Comment:
I suggest adding a comment explaining why we are rate limiting these so future readers don't need to guess, and can avoid adding new per-event info logs.